redis-snapshot-replicator/functions/shipper.py

```diff
@@ -1,28 +1,20 @@
 import boto3
 import botocore
-import datetime
-import re
 import os
 
-iam = boto3.client('iam')  
-instances = os.environ['DB_INSTANCES']  
 target_bucket = os.environ['TARGET_BUCKET'] 
 
 print('Loading function')
 
-def byTimestamp(snap):  
-    if 'SnapshotCreateTime' in snap:
-        return datetime.datetime.isoformat(snap['SnapshotCreateTime'])
-    else:
-        return datetime.datetime.isoformat(datetime.datetime.now())
-
 def lambda_handler(event, context): 
 
     source = boto3.client('elasticache')
 
-    for instance in instances.split(','):
-        source_snaps = source.describe_snapshots(CacheClusterId=instance)['Snapshots']
-        source_snap = sorted(source_snaps, key=byTimestamp, reverse=True)[0]['SnapshotName']
+    message = event['Records'][0]['Sns']['Message']
+
+    if "Snapshot succeeded" in message
+
+        source_snap = ''.join([i for i in message.split('\'') if 'snapshot-' in i])
```

Comment:
Related to my previous comment, this will copy all redis snapshots, regardless of the instance. But actually, we only want to replicate snapshots of specific configured instances. I think you should filter out the snapshots that do not correspond to the configured instances.

Reply:
the sns topic is already configured to receive events only from the required instance. it's a cluster level configuration not entire elasticache events level

Reply:
ok then I understood it wrong 👍